### PR TITLE
Fix interactions with Phoenix 1.2 and Ecto 2.0

### LINKED ISSUE #27: Controller Tests Cruft With Phoenix v1.2.0
Per [upgrade notes](https://gist.github.com/chrismccord/29100e16d3990469c47f851e3142f766#tests), `conn` should now use `build_conn`.

Individual tests using `action` without passing their own `conn` writes the following to stdout:
```
warning: using conn/0 to build a connection is deprecated. Use build_conn/0 instead.
    spec/controllers/registration_controller_spec.exs:2: MyApp.ControllerSpec.action/2
    (espec) lib/espec/let/let_impl.ex:9: ESpec.Let.Impl.let_eval/2
    spec/controllers/registration_controller_spec.exs:2: MyApp.ControllerSpec.should/1
    (espec) lib/espec/example_runner.ex:62: ESpec.ExampleRunner.try_run/3
```

### PR DESCRIPTION
**Pull request by diogobeda**
Fixes #27 and #16, but introduces breaking changes. Because of it, I versioned it as `1.0.0`, but I'd like to hear your opinions on the best approach.

\cc @antonmi

### CODE REVIEW

**Comment by antonmi** [comment]
Hi, Diogo!
Thanks for PR!
Let's make the version 0.3.0. We can add warning about breaking changes to Readme

**Comment by diogobeda** [comment]
Great @antonmi, gonna do it later today. Also will remove `elixir 1.1` from `.travis.yml` as you commented on #26

**Comment by brain-geek** [comment]
@diogobeda Then I'm waiting for your changes to rebase my PR. Because to make mine pass, I need your changes on version updates for phoenix.

**Comment by diogobeda** [comment]
@antonmi I'm kinda insecure on making it `0.3`. According to semver, we'd have to increase the major version when introducing incompatible changes. And with my PR, `phoenix_ecto` will just support `phoenix >= 1.2` cause we're using a function that didn't exist on previous versions.

If you're not comfortable on putting out a `1.0` release now and want to add more features to the release, we can create an `alpha` or `rc` release to wait for those features. What do you think?
It'd not be fair to the lib users to get breaking changes on a minor version.

**Comment by antonmi** [comment]
@diogobeda
ESpecPhoenix is still in the early stage of development. The concept of the library might change in the near future. So I think 0.3.0 is enough

**Comment by diogobeda** [comment]
Okay, then @antonmi. Thanks for the feedback and sorry about the delay, I was just trying to bring up the discussion. If the build passes I think we're good to merge.

**Comment by antonmi** [comment]
Thanks, @diogobeda!
Please remove the last commit, it breaks the build.

File: README.md [hunk 1]
```diff
@@ -67,12 +67,11 @@ ESpec.start
...
 ESpec.configure fn(config) ->
   config.before fn ->
-    #restart transactions
-    Ecto.Adapters.SQL.restart_test_transaction(YourApplication.Repo, [])
+    Ecto.Adapters.SQL.Sandbox.checkout(App.Repo, [])
   end
 
   config.finally fn(_shared) ->
-
+    Ecto.Adapters.SQL.Sandbox(App.Repo, [])
```
Comment (fredwu) [inline README.md]: Typo here, missing `.checkin` ;)

**Comment by antonmi** [comment]
Thanks, guys!

**Pull request merged**